src/shogun/util/factory.h

```diff
@@ -337,5 +334,110 @@ namespace shogun
 	{
 		return std::make_shared<PipelineBuilder>();
 	}
+	} // namespace details
+	template <typename TypeName, typename... Args>
+	std::shared_ptr<TypeName> create(Args... args)
+	{
+		if constexpr (std::is_same_v<TypeName, Features>)
+		{
+			return details::features(std::forward<Args>(args)...);
+		}
+		else if constexpr (std::is_same_v<TypeName, Labels>)
+		{
+			return details::labels(std::forward<Args>(args)...);
+		}
+		else if constexpr (std::is_same_v<TypeName, CSVFile>)
+		{
+			return details::csv_file(std::forward<Args>(args)...);
+		}
+		else if constexpr (std::is_same_v<TypeName, LibSVMFile>)
+		{
+			return details::libsvm_file(std::forward<Args>(args)...);
+		}
+		else if constexpr (std::is_same_v<TypeName, PipelineBuilder>)
+		{
+			return details::pipeline();
+		}
+		else if constexpr (std::is_same_v<TypeName, Kernel>)
+		{
+			return details::kernel(std::forward<Args>(args)...);
+		}
+		else
+		{
+			static_assert(
+			    (sizeof...(Args) == 1) ||
+			        (std::is_constructible_v<Args, std::string> && ...),
+			    "The create function requires a string argument.");
+			return create_object<TypeName>(
+			    (std::string{std::forward<Args>(args)}.c_str())...);
+		}
+	}
+
+	// The SWIG now only support one template parameter in variadic template, so
+	// we have to write those create_ wrapper function
+	std::shared_ptr<Features> create_features(
```

Comment:
Could you hide the templates to fix SWIG's lack of support for variadic templates with `#ifdef SWIG`? There is no need to have this in the C++ library, imo. @vigsterkr?

Reply:
after added `ifdef SWIG`, there are errors that those create_ can't be found